crates/burn-import/src/onnx/dim_inference.rs

```diff
@@ -252,24 +257,40 @@ fn reduce_mean_update_outputs(node: &mut Node) {
 
 /// Update the output tensor dimension based on the "axes" attribute or the second input
 fn unsqueeze_update_output(node: &mut Node) {
-    let axes = match node.inputs.get(1) {
-        Some(input) => match &input.value {
-            Some(Data::Int64s(axes)) => Some(axes.clone()),
-            _ => panic!("Unsqueeze: invalid input types"),
-        },
-        None => node.attrs.get("axes").cloned().map(|v| v.into_i64s()),
+    let axes = if node.inputs.len() == 2 {
+        // get the values while making sure the types are correct
+        match &node.inputs[1].value {
+            Some(value) => match value {
+                Data::Int64s(axes) => Some(axes.clone()),
+                _ => panic!("Unsqueeze: invalid input types"),
+            },
+            None => None,
+        }
+    } else {
+        node.attrs
+            .iter()
+            .find_map(|(key, value)| match key.as_str() {
+                "axes" => Some(value.clone().into_i64s()),
+                _ => None,
+            })
```

Comment:
The changes for unsqueeze look very similar to reshape. If the above refactor works should we apply a similar structure here too?

Reply:
Done